Source/GUI/MemViewer/MemViewer.cpp

```diff
@@ -131,186 +117,275 @@ void MemViewer::changeMemoryRegion(const bool isMEM2)
 {
   m_memViewStart = isMEM2 ? Common::MEM2_START : Common::MEM1_START;
   m_memViewEnd = isMEM2 ? Common::MEM2_END : Common::MEM1_END;
-  verticalScrollBar()->setRange(0, ((m_memViewEnd - m_memViewStart) / 16) - 1);
+  verticalScrollBar()->setRange(0, ((m_memViewEnd - m_memViewStart) / m_numColumns) - m_numRows);
 }
 
 void MemViewer::mousePressEvent(QMouseEvent* event)
 {
+  // Only handle left-click events
+  if (event->button() != Qt::MouseButton::LeftButton)
+    return;
+
   int x = event->pos().x();
   int y = event->pos().y();
-  if (x >= m_rowHeaderWidth && x <= m_rowHeaderWidth + m_hexAreaWidth && y >= m_columnHeaderHeight)
+
+  const bool wasEditingHex = m_editingHex;
+  const int spacing = m_charWidthEm / 2;
+  int clickedHexPosX = x - m_rowHeaderWidth + spacing / 2;
+  int clickedAsciiPosX = x - (m_hexAsciiSeparatorPosX + spacing);
+  int clickedPosY = y - m_columnHeaderHeight;
+
+  // Ignore clicking the headers
+  if (clickedPosY < 0 || clickedHexPosX < 0)
+    return;
+
+  // Ignore space between HEX and ASCII table
+  if (clickedHexPosX > m_hexAreaWidth && clickedAsciiPosX < 0)
+    return;
+
+  // Transform click positions to table positions
+  clickedHexPosX /= spacing + m_charWidthEm * 2;
+  clickedAsciiPosX /= m_charWidthEm;
+  clickedPosY /= m_charHeight;
+
+  // Ignore clicking the padding at the bottom
+  if (clickedPosY >= m_numRows)
+    return;
```

Comment:
I don't like this way of doing this for 2 reasons: 1: it doesn't consider the horizontal space after the ascii area and second, it's honestly more confusing to read to make sure this work because essentially, what you want to do is test that the click was inside the grid and having all these conditions separated doesn't make that part obvious.  I would rather keep the 4 conditions I mentioned because they all set each boundaries of the grid.

Reply:
The area to the right doesn't really need to be checked, since it touches the scroll bar, even when zoomed in very far, but I do agree it would be nicer to do.
I don't agree on the readibility. I'd rather see conditions simply checking two variables than having a bunch of additions within the condition. That's something I'm working on now. The code that you suggested earlier didn't match the areas exactly.
![image](https://user-images.githubusercontent.com/9705046/35010157-41bef2f8-fb02-11e7-81ba-92166b03901b.png)
I'll combine the two, by storing the left/right and top/bottom in variables, then checking if the clicked locaiton is in either of those areas.

Reply:
Ok, sounds fine with better variable labels.

Reply:
I've got the correct area calculations for the hex table now:
```
  int spacing = m_charWidthEm / 2;
  int hexAreaLeft = m_rowHeaderWidth - spacing / 2;
  int hexAreaRight = m_rowHeaderWidth + m_hexAreaWidth - spacing / 2;
  int asciiAreaLeft = (m_hexAsciiSeparatorPosX + spacing);
  int asciiAreaRight = asciiAreaLeft + m_charWidthEm * m_numColumns;
  int AreaTop = m_columnHeaderHeight + m_charHeight - fontMetrics().overlinePos();
  int AreaBottom = AreaTop + m_charHeight * m_numRows;
  int cellWidth = m_charWidthEm * 2 + spacing;
  int cellHeight = m_charHeight;
```
![image](https://user-images.githubusercontent.com/9705046/35011805-c0ad89bc-fb07-11e7-8417-96fca6841913.png)
The click area includes half of the spacing between each byte. Now to get the values for the ascii table and simplyfy it a bit, then I'll get this PR nice and clean! :P